addon/components/frost-checkbox.js

```diff
@@ -76,18 +67,59 @@ export default Component.extend(PropTypeMixin, {
     }
   },
 
-  // == Events ================================================================
+  // == DOM Events ============================================================
 
+  /**
+   * Handle the focusIn event
+   * @params {Event} e - the event being handled
+   */
   _onFocus: Ember.on('focusIn', function (e) {
     // If an onFocus handler is defined, call it
     if (this.attrs.onFocus) {
       this.attrs.onFocus()
     }
   }),
 
+  /**
+   * Handle the keypress event
+   * @param {Event} e - the event being handled
+   * @returns {Boolean|undefined} false if we want to stop propagation
+   */
+  keyPress (e) {
+    if (e.keyCode === 32) {
+      if (get(this, 'disabled') !== true) {
+        this.$('input').prop('checked', !this.$('input').prop('checked'))
+        this.send('input')
+      }
+      e.preventDefault()
+      e.stopPropagation()
+      return false
+    }
+  },
+
+  // == Lifecycle Hooks =======================================================
+
+  /* Ember.Component method */
```

Comment:
I'm sure we're not debating wether or not this is jsdoc (you're right) - the point in this case would be that the section header speaks for the elements within it.  If we're going with inline comments then I would expect a description of what is being done in the hook moreso than a description of the hook.  Another one for the inline docs on all functions debate.